### ISSUE TITLE
`semver` fails on valid versions

### ISSUE THREAD

**Issue by typetetris** [fosskers/versions#51]
Hi!

With versions version 5.0.0, I get the following behavior:
```
*Data.Versions> let showVer = putStrLn . either errorBundlePretty (Data.Text.unpack . prettySemVer)  . semver
*Data.Versions> showVer "1.2.2-00a"
Semantic Version:1:8:
  |
1 | 1.2.2-00a
  |        ^
unexpected '0'
expecting '+', '.', or end of input

*Data.Versions> showVer "1.2.2-0a"
1.2.2-0a
*Data.Versions>
```

**Comment by typetetris** [comment]
And btw: Thanks for that awesome package!

**Comment by fosskers** [comment]
You're welcome!

Well unless I'm [reading this](https://semver.org/#spec-item-9) incorrectly, numeric segments of prerelease sections may not contain leading zeroes. So, neither of those should parse as `SemVer`.

**Comment by typetetris** [comment]
I think you are misreading this.

In the same document a little further down, there is a [grammar](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions). The grammer clearly states numeric identifiers are made up only of decimal digits.

The versions above contain the prerelease identifier "0a" and "00a" which contain "a", so they are alphanumeric identifiers which have no requirement about leading zeros.

**Comment by fosskers** [comment]
Thanks for the clarification. This seems to be failing for the Rust version as well, while the `semver` crate accepts it. I'm quite busy next week, but have scheduled the fix (for the Rust version first) for Friday.

**Comment by fosskers** [comment]
After analysing the grammar more, it looks like I need to redo how the `Chunk` and `Chunks` types work entirely.

**Comment by fosskers** [comment]
I've fixed this on the Rust side and will get to the Haskell when I can.

**Comment by fosskers** [comment]
I just released `6.0.0`, which includes a fix for this. Sorry for the wait!
